app/services/AppealService.scala

```diff
@@ -125,4 +125,17 @@ class AppealService @Inject()(appealsConnector: PEGAConnector,
       None
     }
   }
+
+  def truncateFilename(fileName: String): String = {
+    logger.info(s"[AppealService][truncateFilename] File name length: ${fileName.length}, truncating to ${appConfig.maximumFilenameLength}")
+    if(fileName.contains(".")) {
+      val fileNameSplit = fileName.split("\\.")
+      val fileNameMain = fileNameSplit.head
```

Comment:
I haven't tested this myself but this may be an incorrect assumption. How would this react if we had a file name: 'this.is.a.file.txt'?

Reply:
I have seen files being uploaded where . has been used for date separators for example so this will need a re think
`VAT Return 07.04.21.pdf`

Reply:
I could probably wrangle the regex to find the last . and split it from that

Reply:
I think rather than doing `.head` you could drop the last element (which would be the file extension)

Reply:
Not sure what order this will be done but if it is before the file extension adding we will have to be careful of cases where it is missing, just something else to consider.

Reply:
We have a list of accepted file types, right? Could we maybe we could check against their extensions?